native-activity/app/src/main/AndroidManifest.xml

```diff
@@ -4,7 +4,13 @@
     android:versionCode="1"
           android:versionName="1.0">
 
-  <!-- This .apk has no Java code itself, so set hasCode to false. -->
+  <!--
+  This .apk has no Java/Kotlin code, so set hasCode to false.
```

Comment:
i'm assuming you _have_ to set this to `false` for this to work at all? is it worth saying _why_?

Reply:
<img width="829" alt="截屏2024-04-17 上午9 53 41" src="https://github.com/android/ndk-samples/assets/68465898/eef40f4e-5d8b-4cd9-8ac3-71cff3014385">
hi DanAlbert and enh-google, I hold the different view.
set hasCode to false just for bytecode optimized, it is not ‘must have’, and it also can be a trap when someone want to add Jave code. especially for engineers who are not app developers. 

explain why this error: https://github.com/android/ndk-samples/pull/998

Reply:
> i'm assuming you have to set this to false for this to work at all? is it worth saying why?

My best guess is that it improves app startup time, but I'm still waiting to hear from anyone that can say with certainty what it does. hackbod wrote the original comment rather than removing the property, so I assume it is worth keeping.

Reply:
(resolving so I can merge since this is definitely not worse, I'll be back with another PR if I learn that it should be removed)

Reply:
No, it should not be removed. Setting it incorrectly has no symptoms on the device I tested with, but it may on others, and may in the future (which is how this problem cropped up in the first place, the bug was asymptomatic for a time and then wasn't, we shouldn't repeat that in the other direction): https://developer.android.com/guide/topics/manifest/application-element#code

Reply:
I don't think it would has symptoms on the other devices or in the future, but it definitely not worse. so I agree with you. 🍺 🍻